sys/include/net/ng_ipv6/addr.h

```diff
@@ -282,6 +346,30 @@ static inline bool ng_ipv6_addr_is_link_local(const ng_ipv6_addr_t *addr)
 }
 
 /**
+ * @brief   Checks if @p addr is a site-local address.
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.5.7">
+ *          RFC 4291, section 2.5.7
+ *      </a>
+ *
+ * @note Site-local addresses are now deprecated as defined in <a
+ *       href="http://tools.ietf.org/html/rfc3879">SLDEP</a>. They are only
+ *       defined here for the distinction of global unicast addresses.
+ *
+ * @param[in] addr  An IPv6 address.
+ *
+ * @return  true, if @p addr is a site-local unicast address,
+ * @return  false, otherwise.
+ */
+static inline bool ng_ipv6_addr_is_site_local(const ng_ipv6_addr_t *addr)
+{
+    return (((byteorder_ntohs(addr->u16[0]) & 0xFFC0) ==
+             NG_IPV6_ADDR_SITE_LOCAL_PREFIX) ||
```

Comment:
No diff for the buildsize on all platforms except x86 (native and qemu) where this change introduced ~150 Bytes more code.